### PR TITLE
Support for RHEL image builder on vSphere

### PR DESCRIPTION
**Pull request by vignesh-goutham**
*Issue #, if available:*  #1298

*Description of changes:*
Adds support for building RHEL images for vSphere

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/aws/image-builder/builder/builder.go [hunk 1]
```diff
@@ -125,27 +145,41 @@ func (b *BuildOptions) BuildImage() {
...
-	b.VsphereConfig, err = filepath.Abs(b.VsphereConfig)
-	if err != nil {
-		log.Fatalf("Error converting vsphere config path to absolute")
-	}
+	// Validate vsphere config inputs
+	if b.VsphereConfig != nil {
+		// Validate Rhel username and password
+		if b.Os == RedHat {
+			if b.VsphereConfig.RhelUsername == "" || b.VsphereConfig.RhelPassword == "" {
+				log.Fatalf("\"rhel_username\" and \"rhel_username\" are required fields in vsphere-config when os is redhat")
```
Comment (gwesterfieldjr) [inline projects/aws/image-builder/builder/builder.go:166]: Looks like you doubled rhel_username in the log message.

File: projects/aws/image-builder/builder/types.go [hunk 2]
```diff
@@ -2,15 +2,38 @@ package builder
...
+	Network            string `json:"network"`
+	ResourcePool       string `json:"resource_pool"`
+	Template           string `json:"template"`
+	VcenterServer      string `json:"vcenter_server"`
+	Username           string `json:"username"`
+	Password           string `json:"password"`
+	IsoUrl             string `json:"iso_url,omitempty"`
+	IsoChecksum        string `json:"iso_checksum,omitempty"`
+	IsoChecksumType    string `json:"iso_checksum_type,omitempty"`
+	RhelUsername       string `json:"rhel_username"`
```
Comment (gwesterfieldjr) [inline projects/aws/image-builder/builder/types.go:37]: Just curious why vsphereconfig has a rhel username/password? Maybe rename it to OsUser and OsPassword so its more generic and we can leverage it for other OS templates in the future as needed.

File: projects/aws/image-builder/builder/builder.go [hunk 3]
```diff
@@ -41,24 +42,43 @@ func (b *BuildOptions) BuildImage() {
...
-		err = executeMakeBuildCommand(buildCommand, b.ReleaseChannel)
+
+		var buildCommand string
+		switch b.Os {
+		case Ubuntu:
+			buildCommand = fmt.Sprintf("make -C %s local-build-ova-ubuntu-2004", imageBuilderProjectPath)
+		case RedHat:
+			buildCommand = fmt.Sprintf("make -C %s local-build-ova-rhel-7", imageBuilderProjectPath)
+			commandEnvVars = append(commandEnvVars,
+				fmt.Sprintf("RHSM_USERNAME=%s", b.VsphereConfig.RhelUsername),
```
Comment (gwesterfieldjr) [inline projects/aws/image-builder/builder/builder.go:76]: Ah I see we pass it here to the image-builder target. What does image builder do with it?

Comment (abhay-krishna) [reply projects/aws/image-builder/builder/builder.go:76]: These values need to be passed in and set as the values for RHSM_USER and RHSM_PASS env vars for the Ansible step of RedHat subscription - https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/ansible/roles/setup/tasks/redhat.yml#L15-L21

File: projects/aws/image-builder/builder/types.go (hunk 2 above)
Comment (vignesh-goutham) [reply projects/aws/image-builder/builder/types.go:37]: Yea its actually for redhat subscription manager. This is not the regular linux username/password

File: projects/aws/image-builder/builder/builder.go (hunk 3 above)
Comment (vignesh-goutham) [reply projects/aws/image-builder/builder/builder.go:76]: It's required to register with redhat subscription manager to update the image.

**Review by abhay-krishna — APPROVED** [review]
/lgtm

**Pull request merged**